src/main/java/edu/ucsb/cs56/mapache_search/GoogleSearchService.java

```diff
@@ -25,7 +25,11 @@
     private String searchId = "001539284272632380888:kn5n6ubsr7x";
     private String apiKey; // TODO: set API key per user somehow
 
-    public GoogleSearchService(@Value("${google.search.api.key}") String apiKey) {
+    public GoogleSearchService(@Value("${google.search.api.key:}") String apiKey) {
+        if (apiKey.isEmpty()) {
+            throw new IllegalStateException("dude, you need an api key in your localhost.json, kthxbye.");
+        }
```

Comment:
Nice! 

Though in all seriousness, the error message should probably be something such as:

```
google.search.api.key is not defined.   Check the README.md for more instructions.
```

And then maybe we put into the README.md something like:

```
If you get an `IllegalStateException` with the message
`google.search.api.key is not defined` you should take
these steps if running on `localhost`:
  * check that  `google.search.api.key` is defined in `localhost.json`
  * check that you did `source env.sh`
```

There should eventually be instructions for Heroku too, but that can be a separate issue.